From 69e1ac499e7b36e4a48068c7ca5dcdd1ea7848c3 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <rchintakuntla@cavium.com>
Date: Wed, 20 Aug 2014 15:10:58 -0700
Subject: [PATCH 06/18] arm64: gicv3: its: Increase FORCE_MAX_ZONEORDER for
 Cavium ThunderX

Patch-mainline: Submitted, http://lists.infradead.org/pipermail/linux-arm-kernel/2015-May/340461.html
References: bsc#1105816

In case of ARCH_THUNDER, there is a need to allocate the GICv3 ITS table
which is bigger than the allowed max order. So we are forcing it only in
case of 4KB page size.

Signed-off-by: Radha Mohan Chintakuntla <rchintakuntla@cavium.com>
[rric: use ARM64_4K_PAGES since we have now ARM64_16K_PAGES, change order]
Signed-off-by: Robert Richter <rrichter@cavium.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm64/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -591,6 +591,7 @@ config FORCE_MAX_ZONEORDER
 	int
 	default "14" if (ARM64_64K_PAGES && TRANSPARENT_HUGEPAGE)
 	default "12" if (ARM64_16K_PAGES && TRANSPARENT_HUGEPAGE)
+	default "13" if (ARM64_4K_PAGES && ARCH_THUNDER)
 	default "11"
 	help
 	  The kernel memory allocator divides physically contiguous memory
